[LB] Break out best practices into separate how-to guides (#2033)
cloudflare/cloudflare-docs: products/load-balancing/src/content/additional-options/planned-maintenance.md

---
order: 5
pcx-content-type: how-to
---

# Planned maintenance

When you change application settings or add new assets, you will likely want to make these changes on one origin server at a time. Going server by server reduces the risk of changes and ensures a more consistent user experience.

> To take servers out of rotation **gradually** (important for session-based load balancing), [enable origin drain](#gradual-rotation) on your load balancer. This option is only available for [proxied load balancers (orange-clouded)](/understand-basics/proxy-modes).

Reviewer: ```suggestion
To take servers out of rotation gradually (important for session-based load balancing), [enable origin drain](#gradual-rotation) on your load balancer. This option is only available for [proxied load balancers (orange-clouded)](/understand-basics/proxy-modes).
```

> To direct traffic away from origin server **immediately**, [adjust settings on the pool or monitor](#immediate-rotation).

Reviewer: ```suggestion
To direct traffic away from your origin server immediately, [adjust settings on the pool or monitor](#immediate-rotation).
```

<Aside type="note">

If you want to divert traffic from an origin to prevent it from becoming unhealthy, use [Load Shedding](/additional-options/load-shedding) instead.

</Aside>

## Before you begin

Before disabling any origin server, review the settings for any affected load balancers and pools. 

> If a pool falls below it's **Health Threshold**, it will be considered **Unhealthy** and — depending on the load balancer setup and steering policy — a load balancer may begin routing traffic away from that pool.

Reviewer: ```suggestion
If a pool falls below its **Health Threshold**, it will be considered **Unhealthy** and — depending on the load balancer setup and steering policy — a load balancer may begin routing traffic away from that pool.
```

## Gradual rotation

<Aside type="note">

Origin drain is only available for [proxied load balancers (orange-clouded)](/understand-basics/proxy-modes).

</Aside>

> With [session-based load balancing](/understand-basics/session-affinity), it's important to direct all requests from a particular end user to a specific origin server. Otherwise, information about the user session — such as items in their shopping cart — may be lost and lead to negative business outcomes.

Reviewer: ```suggestion
With [session-based load balancing](/understand-basics/session-affinity), it is important to direct all requests from a particular end user to a specific origin server. Otherwise, information about the user session — such as items in their shopping cart — may be lost and lead to negative business outcomes.
```
